gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/RemoveRemoteVoltageRegulationInFranceService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.swe.runner.app.services;
+
+import com.farao_community.farao.swe.runner.app.configurations.DataFixConfiguration;
+import com.powsybl.iidm.network.Country;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.Substation;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+/**
+ * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
+ */
+@Service
+public class RemoveRemoteVoltageRegulationInFranceService {
+
+    private final DataFixConfiguration dataFixConfiguration;
+    private final Logger businessLogger;
+
+    public RemoveRemoteVoltageRegulationInFranceService(DataFixConfiguration dataFixConfiguration, Logger businessLogger) {
+        this.dataFixConfiguration = dataFixConfiguration;
+        this.businessLogger = businessLogger;
+    }
+
+    public void removeRemoteVoltageRegulationInFrance(Network network) {
+        if (!dataFixConfiguration.removeRemoteVoltageRegulationInFrance()) {
+            return;
+        }
+        businessLogger.info("Removing remote voltage regulations in France.");
+        network.getGeneratorStream()
+                .filter(this::isRemoteVoltageRegulationOn)
+                .filter(this::isInFrance)
+                .forEach(this::removeRemoteVoltageRegulation);
+    }
+
+    private void removeRemoteVoltageRegulation(Generator generator) {
+        double oldVoltageTargetInPu = generator.getTargetV() / generator.getRegulatingTerminal().getVoltageLevel().getNominalV();
+        double newVoltageTarget = oldVoltageTargetInPu * generator.getTerminal().getVoltageLevel().getNominalV();
+        generator.setTargetV(newVoltageTarget);
+        generator.setRegulatingTerminal(generator.getTerminal());
+        businessLogger.info(String.format("Deactivating remote voltage regulation of generator '%s' new local target set to %fkV", generator.getId(), newVoltageTarget));
+    }
+
+    private boolean isInFrance(Generator generator) {
```

Comment:
This method could be written with `map`/`flatMap` methods, but this may decrease readability:
```java
return generator.getTerminal().getVoltageLevel().getSubstation()
        .flatMap(Substation::getCountry)
        .map(country -> country == Country.FR)
        .orElse(false);
```

Reply:
This is exactly the reason why I put it as it is, and it is more complicated than that, as it both getSubstation and getCountry returns Optional and not Country directly. The best would be to be able to impelment it directly in powsybl-core and allow to call something like generator.getCountry() that would return an Optional